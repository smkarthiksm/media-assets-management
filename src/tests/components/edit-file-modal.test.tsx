import '@testing-library/jest-dom';
import { renderWithProviders } from '../test-utility';
import EditFileModalComponent from '../../components/edit-file-modal/edit-file-modal';
import { fireEvent, screen } from '@testing-library/react';

describe('EditFileModalComponent', () => {
  test('should render', () => {
    const { container } = renderWithProviders(
      <EditFileModalComponent
        open={true}
        handleCancel={() => jest.fn()}
        handleSave={() => jest.fn()}
      />,
    );
    expect(container).toMatchSnapshot();
  });

  test(`should update first record's fields on text field change`, () => {
    const { store } = renderWithProviders(
      <EditFileModalComponent
        open={true}
        handleCancel={() => jest.fn()}
        handleSave={() => jest.fn()}
      />,
      {
        preloadedState: {
          editFileModal: {
            index: 0,
            isEditFileModalVisible: true,
            title: 'testTitle1',
          },
        },
      },
    );

    const element = screen.getByRole('textbox', {
      name: 'Title',
    }) as HTMLTableCellElement;

    expect(store.getState().editFileModal.title).toEqual('testTitle1');
    fireEvent.change(element, { target: { value: 'updatedTitle' } });
    expect(store.getState().editFileModal.title).toEqual('updatedTitle');
  });
});
