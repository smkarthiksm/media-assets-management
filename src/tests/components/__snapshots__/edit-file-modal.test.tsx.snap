// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AllFilesHeaderComponent should render 1`] = `
<div>
  <div
    class="body-container h-100"
  />
</div>
`;

exports[`EditFileModalComponent should render 1`] = `
<div
  aria-hidden="true"
/>
`;

exports[`EditFileModalComponent should update fields on text field change 1`] = `
<div
  aria-hidden="true"
/>
`;
